status-im/status-react
# refactor: suggested routes view

fixes #19696 

### Summary

This PR refactors suggested network routes view. Please read the issue for context on the why we needed to refactor the component, but basically it wasn't complying with different cases.

This new approach enables a more flexible UI where different types of network links components are used to connect network summary cards in the suggested routes screen. Some examples:

<img width="350" alt="Screenshot 2024-04-23 at 4 20 05 PM" src="https://github.com/status-im/status-mobile/assets/18485527/a06859be-8a22-43d7-8460-7ee34cac9dea">
<img width="350" alt="Screenshot 2024-04-23 at 4 25 46 PM" src="https://github.com/status-im/status-mobile/assets/18485527/9456c639-0879-42e3-9501-98df005f698f">
<img width="350" alt="Screenshot 2024-04-23 at 4 27 42 PM" src="https://github.com/status-im/status-mobile/assets/18485527/b3ae2f23-69a1-4819-a563-447c3470b20f">
<img width="350" alt="Screenshot 2024-04-23 at 4 29 11 PM" src="https://github.com/status-im/status-mobile/assets/18485527/e9a4cef3-f529-4e5c-a8f2-4c77a999d059">
<img width="350" alt="Screenshot 2024-04-23 at 4 29 53 PM" src="https://github.com/status-im/status-mobile/assets/18485527/e1247659-cf2c-49f9-b990-484c8e70d612">

At the same time it adds some new flows for routes view:

- User can't disable last network: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=13183-162514&mode=design&t=38A1QPnw43BzXEYs-4
- No routes found (empty response from algorithm): https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=13196-166879&mode=design&t=38A1QPnw43BzXEYs-4

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

Smoke test:
- Open Status
- Log in
- Go to wallet
- Select account
- Tap on Send
- Select a recipient
- Select a token
- Enter a valid amount
- Check routes are being rendered correctly

Disable last chain:
- Open Status
- Log in
- Go to wallet
- Select account
- Tap on Send
- Select a recipient
- Select a token
- Enter a valid amount
- Disable all chains until there is only one chain left
- Verify the user can't disable the last chain and the correct message is shown

No routes found:
- Open Status
- Log in
- Go to wallet
- Select account
- Tap on Send
- Select a recipient
- Select a token
- Enter a valid amount until the responde from status-go is empty
- Verify a "No routes found" message is shown to the user with a `Try again` button
- User could click the `Try again` button to search for routes again

status: ready
